scripts/tx-builder.js

```diff
@@ -0,0 +1,208 @@
+const { TransactionBlock } = require('@mysten/sui.js/transactions');
+const { bcs } = require('@mysten/bcs');
+const {
+    utils: { arrayify, hexlify },
+} = require('ethers');
+
+function getObject(tx, object) {
+    if (Array.isArray(object)) {
+        object = hexlify(object);
+    }
+
+    if (typeof object === 'string') {
+        return tx.object(object);
+    }
+
+    return object;
+}
+
+function getTypeName(type) {
+    function get(type) {
+        let name = `${type.address}::${type.module}::${type.name}`;
+
+        if (type.typeArguments.length > 0) {
+            name += `<${type.typeArguments[0]}`;
+
+            for (let i = 1; i < type.typeArguments.length; i++) {
+                name += `,${type.typeArguments[i]}`;
+            }
+
+            name += '>';
```

Comment:
i think this part can be shorten using `join(",")` function e.g. name += `<${type.typeArguments.join(",")}>`